File: frontend/src/types/comment.ts
/**
 * Comment Types - 评论类型定义
 */

export interface Comment {
  id: number
  content: string
  html_content?: string
  issue_id: number
  author_id: number
  author?: UserLite
  parent_id?: number
  is_resolved: boolean
  resolved_by_id?: number
  resolved_at?: string
  reaction_count: number
  replies?: Comment[]
  created_at: string
  updated_at: string
}

export interface CommentCreate {
  issue_id: number
  content: string
  html_content?: string
  parent_id?: number
}

export interface CommentUpdate {
  content?: string
  html_content?: string
}

export interface CommentListResponse {
  items: Comment[]
  total: number
  page: number
  page_size: number
}

export interface UserLite {
  id: number
  username: string
  display_name?: string
  email: string
  avatar_url?: string
}

File: backend/app/schemas/comment.py
"""
Comment Schemas - 评论Schema定义
"""
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, Field


class CommentBase(BaseModel):
    """评论基础Schema"""
    content: str
    html_content: Optional[str] = None


class CommentCreate(CommentBase):
    """创建评论"""
    issue_id: int
    parent_id: Optional[int] = None


class CommentUpdate(BaseModel):
    """更新评论"""
    content: Optional[str] = None
    html_content: Optional[str] = None


class CommentResponse(CommentBase):
    """评论响应"""
    id: int
    issue_id: int
    author_id: int
    parent_id: Optional[int] = None
    is_resolved: bool
    resolved_by_id: Optional[int] = None
    resolved_at: Optional[datetime] = None
    reaction_count: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class CommentResolve(BaseModel):
    """解决评论"""
    is_resolved: bool = True


class CommentListResponse(BaseModel):
    """评论列表响应"""
    items: List[CommentResponse]
    total: int
    page: int
    page_size: int
